packages/connect-form/src/components/advanced-options-tabs/advanced-tab/url-options-table.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { ChangeEvent, useEffect } from 'react';
+import { css } from '@emotion/css';
+import {
+  spacing,
+  Table,
+  TableHeader,
+  Row,
+  Cell,
+  Button,
+  IconButton,
+  Icon,
+  TextInput,
+  Select,
+  Option,
+  OptionGroup,
+  Banner,
+} from '@mongodb-js/compass-components';
+import ConnectionStringUrl from 'mongodb-connection-string-url';
+
+import { editableUrlOptions, UrlOption } from '../../../utils/url-options';
+import { UpdateConnectionFormField } from '../../../hooks/use-connect-form';
+
+const tableHeaderStyles = css({
+  width: '50%',
+});
+
+const optionValueCellStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  width: '100%',
+});
+
+const addUrlOptionsButtonStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[3],
+  marginBottom: spacing[2],
+});
+
+function getUrlOptions(connectionStringUrl: ConnectionStringUrl): UrlOption[] {
+  let opts: string[] = [];
+  const urlOptions: UrlOption[] = [];
+
+  editableUrlOptions.forEach(({ values }) => (opts = opts.concat(values)));
+  connectionStringUrl.searchParams.forEach((value, name) => {
```

Comment:
Would it help to use `.typedSearchParams<MongoClientOptions>()` here?
Here's an example:
https://github.com/mongodb-js/compass/blob/main/packages/connect-form/src/components/advanced-options-tabs/general/direct-connection-input.tsx#L17
Maybe we can have these UrlOptions be a partial of `import type { MongoClientOptions } from 'mongodb';`